Automattic/wp-calypso
# Migrate App Passwords from Redux to React Query

#### Changes proposed in this Pull Request

Started by removing the `UNSAFE_componentWillReceiveProps` found in `<ApplicationPasswords />` I took the opportunity to also migrate it to React Query and remove all related Redux and data layer code. 

* Move Application Passwords (shown at `/me/security/two-step`) to use React Query

`<ApplicationsPasswords />` used to be one big component which I split up into smaller components by extracting `renderX` methods into their own smaller components.

#### Testing instructions

* Go to `/me/security/two-step`
* Make sure _Application Passwords_ work exactly like on prod (create and delete passwords, make sure the list changes accordingly and the form behaves as expected)
* Make sure to test error cases by blocking network requests to
  * `/me/two-step/application-passwords/new`
  * `/me/two-step/application-passwords/*/delete`

Also related to #58453
